src/solvers/refinement/string_refinement.cpp

```diff
@@ -718,25 +696,11 @@ decision_proceduret::resultt string_refinementt::dec_solve()
 
       if(index_sets.current.empty())
       {
-        debug() << "current index set is empty" << eom;
-        if(axioms.not_contains.empty())
-        {
-          debug() << "no not_contains axioms, hence SAT" << eom;
-          concretize_lengths(
-            found_length,
-            get,
-            symbol_resolve,
-            generator.get_created_strings());
-          return resultt::D_SATISFIABLE;
-        }
-        else
-        {
-          debug() << "not_contains axioms exist, hence ERROR" << eom;
-          return resultt::D_ERROR;
-        }
+        error() << "dec_solve: current index set is empty, "
+                << "this should not happen" << eom;
+        return resultt::D_ERROR;
```

Comment:
@romainbrenguier Changed the behavior of the solver here, probably not on purpose. In a later commit to revert to what was originally there for the most part, but not completely.